File: api/user-service/src/main/java/com/carrental/userservice/model/dto/PasswordUpdateDTO.java
package com.carrental.userservice.model.dto;

import com.carrental.userservice.annotation.PasswordMatches;
import com.carrental.userservice.annotation.ValidPassword;
import lombok.Data;
import org.hibernate.validator.constraints.Length;

import javax.validation.constraints.NotEmpty;

@Data
@PasswordMatches
public class PasswordUpdateDTO {

    @NotEmpty
    @Length(max = 16)
    private String currentPassword;

    @NotEmpty
    @ValidPassword
    @Length(max = 16)
    private String newPassword;

    @NotEmpty
    @Length(max = 16)
    private String confirmPassword;
}

File: frontend-app/shared/model/PasswordUpdateDTO.ts
export class PasswordUpdateDTO {
    currentPassword: string;
    newPassword: string;
    confirmPassword: string;

    constructor(currentPassword: string, newPassword: string, confirmPassword: string) {
        this.currentPassword = currentPassword;
        this.newPassword = newPassword;
        this.confirmPassword = confirmPassword;
    }
}
